File: src/main/java/com/refinedmods/refinedstorage/container/slot/legacy/LegacyFilterSlot.java
package com.refinedmods.refinedstorage.container.slot.legacy;


import net.minecraft.world.Container;
import net.minecraft.world.entity.player.Player;
import net.minecraft.world.item.ItemStack;

import javax.annotation.Nonnull;

public class LegacyFilterSlot extends LegacyBaseSlot {
    public LegacyFilterSlot(Container inventory, int inventoryIndex, int x, int y) {
        super(inventory, inventoryIndex, x, y);
    }

    @Override
    public boolean mayPickup(Player player) {
        return false;
    }

    @Override
    public boolean mayPlace(ItemStack stack) {
        return true;
    }

    @Override
    public void set(@Nonnull ItemStack stack) {
        if (!stack.isEmpty()) {
            stack.setCount(1);
        }

        super.set(stack);
    }
}

File: src/main/kotlin/com/refinedmods/refinedstorage/container/slot/legacy/LegacyFilterSlot.kt
package com.refinedmods.refinedstorage.container.slot.legacy

import com.refinedmods.refinedstorage.api.network.grid.IGrid.slotId
import net.minecraft.entity.player.PlayerEntity
import net.minecraft.inventory.IInventory
import net.minecraft.item.ItemStack

class LegacyFilterSlot(inventory: IInventory?, inventoryIndex: Int, x: Int, y: Int) : LegacyBaseSlot(inventory, inventoryIndex, x, y) {
    fun canTakeStack(player: PlayerEntity?): Boolean {
        return false
    }

    fun isItemValid(stack: ItemStack?): Boolean {
        return true
    }

    fun putStack(@Nonnull stack: ItemStack) {
        if (!stack.isEmpty) {
            stack.count = 1
        }
        super.putStack(stack)
    }
}